sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InferWindowGroupLimit.scala

```diff
@@ -68,10 +72,57 @@ object InferWindowGroupLimit extends Rule[LogicalPlan] with PredicateHelper {
     case _ => false
   }
 
+  private def limitSupport(limit: Int, window: Window): Boolean =
+    limit <= conf.windowGroupLimitThreshold && window.child.maxRows.forall(_ > limit) &&
+      !window.child.isInstanceOf[WindowGroupLimit] &&
+      window.windowExpressions.forall(isExpandingWindow) &&
+      window.orderSpec.exists(!_.foldable) &&
```

Comment:
Why need this check? What will happen if all the `orderSpec` are foldable.

Reply:
If `orderSpec` are foldable, equivalent orderSpec is `Nil`, function calculation requires all data of each window group.

Reply:
Sounds reasonable.

Reply:
After a second thought, shall we allow all the `orderSpec` are foldable if all the window expressions with `RowFrame`?

Reply:
yea, I didn't think it through, I'll make changes right away.